language/libra-vm/src/libra_vm.rs

```diff
@@ -505,6 +528,98 @@ impl LibraVM {
         })
     }
 
+    fn process_writeset_transaction(
+        &self,
+        remote_cache: &mut BlockDataCache<'_>,
+        txn: SignedTransaction,
+    ) -> VMResult<TransactionOutput> {
+        let txn = match txn.check_signature() {
+            Ok(t) => t,
+            _ => {
+                return Ok(discard_error_output(VMStatus::new(
+                    StatusCode::INVALID_SIGNATURE,
+                )))
+            }
+        };
+
+        let change_set = if let TransactionPayload::WriteSet(change_set) = txn.payload() {
+            change_set
+        } else {
+            // TODO: Should we return errors to the caller?
```

Comment:
Better TODO? What errors should we consider returning/why have we chosen not to?